include/counter.h

```diff
@@ -460,6 +501,71 @@ static inline u32_t z_impl_counter_get_max_relative_alarm(struct device *dev)
 	return api->get_max_relative_alarm(dev);
 }
 
+/**
+ * @brief Set guard period in ticks.
+ *
+ * Setting non-zero guard period enables detection of setting absolute alarm
+ * too late. It limits how far in the future absolute alarm can be set.
+ *
+ * Detection of too late setting is vital since if it is not detected alarm
+ * is delayed by full period of the counter (up to 32 bits). Because of the
+ * wrapping, it is impossible to distinguish alarm which is short in the past
+ * from alarm which is targeted to expire after full counter period. In order to
+ * detect too late setting, longest possible alarm is limited. Absolute value
+ * cannot exceed: (now + top_value - guard_period) % top_value.
+ *
+ * Guard period depends on application and counter frequency. If it is expected
+ * that absolute alarms setting might be delayed then guard period should
+ * exceed maximal potential delay. If use case allows, guard period can be set
+ * very high (e.g. half of the counter top value).
+ *
+ * After initialization guard period is set to 0 and late detection is disabled.
+ *
+ * @param dev		Pointer to the device structure for the driver instance.
+ * @param guard		Guard period in counter ticks.
```

Comment:
Since we are going to update this PR anyway, what about using `period` or `guard_period` instead of `guard` as a parameter name. Guard seems too ambiguous/misleading.

Reply:
> what about using `period` or `guard_period` instead of `guard` as a parameter name.

Or maybe `ticks`?

Reply:
changed to ticks